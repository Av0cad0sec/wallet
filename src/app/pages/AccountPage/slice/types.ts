import { AccountsRow, OperationsRow } from 'vendors/explorer'

export interface BalanceDetails {
  total: number
  debonding: number
  escrow: number
  available: number
  delegations: number
}

/* --- STATE --- */
export interface AccountState extends AccountsRow {
  loading: boolean
  transactions: OperationsRow[]
}
