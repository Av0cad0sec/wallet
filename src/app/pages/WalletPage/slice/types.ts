import { ErrorPayload } from 'types/errors'

export enum TransactionType {
  SENT = 0,
  RECEIVED = 1,
}

export enum WalletType {
  Ledger = 'ledger',
  PrivateKey = 'private_key',
  Mnemonic = 'mnemonic',
}

export interface Transaction {
  address: string
  date: number
  amount: number
  type: TransactionType
}

export interface TransactionSent {
  from: string
  to: string
  amount: number
}
export interface SendTransactionPayload {
  /* bech32 Address */
  to: string

  /* Token amount */
  amount: number
}

/**
 * WalletBalance
 *
 * We stock the amounts as strings to work around
 * poor bigint support in JS
 *
 */
export interface WalletBalance {
  available: string
  escrow: string
  debonding: string
}

export interface Wallet {
  publicKey: string
  address: string
  type?: WalletType
  privateKey?: string
}

export interface TransactionStatus {
  success: boolean
  error?: ErrorPayload
}

/* --- STATE --- */
export interface WalletState extends Wallet {
  isOpen: boolean
  balance: WalletBalance
  transaction: TransactionStatus
}
