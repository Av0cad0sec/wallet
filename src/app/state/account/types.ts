import { Transaction } from 'app/state/transaction/types'
import { ErrorPayload } from 'types/errors'

export interface BalanceDetails {
  available: number | null
  debonding: number | null
  delegations: number | null
  total: number | null
}

export interface Account extends BalanceDetails {
  address: string
}

/* --- STATE --- */
export interface AccountState extends Account {
  loading: boolean
  accountError?: ErrorPayload
  transactions: Transaction[]
  transactionsError?: ErrorPayload
}
