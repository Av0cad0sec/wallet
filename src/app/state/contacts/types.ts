export interface Contact {
  address: string
  name: string
}

/* --- STATE --- */
export interface ContactsState {
  [address: string]: Contact
}
