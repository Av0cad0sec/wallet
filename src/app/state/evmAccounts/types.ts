export interface EvmAccount {
  ethPrivateKey: string
  ethAddress: string
}

/* --- STATE --- */
export interface EvmAccounts {
  [ethAddress: string]: EvmAccount
}
