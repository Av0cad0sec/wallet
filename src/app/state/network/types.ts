/* --- STATE --- */

export type NetworkType = 'local' | 'testnet' | 'mainnet'

export interface NetworkState {
  selectedNetwork: NetworkType

  /** Ticker symbol (ROSE, TEST, etc.) */
  ticker: string

  /** chainContext / Genesis Hash */
  chainContext: string
}
