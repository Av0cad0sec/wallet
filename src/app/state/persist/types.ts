import { RootState } from 'types'

type StrictOmit<T, K extends keyof T> = Omit<T, K>

interface PreserveAliasName extends String {}

export interface KeyWithSalt {
  key: Uint8Array
  salt: Uint8Array
}
/** Redux can't serialize Uint8Arrays so we base64 encode and stringify */
export type StringifiedKeyWithSalt = string & PreserveAliasName

export interface EncryptedObject {
  secretbox: Uint8Array
  nonce: Uint8Array
  salt: Uint8Array
}
/** Redux can't serialize Uint8Arrays so we base64 encode and stringify */
export type EncryptedString = string & PreserveAliasName
// Hover to check if inferred variable type is EncryptedString (not string)
export const testPreserveAliasName = '{}' as EncryptedString

/* --- STATE --- */
export interface PersistState {
  hasPersistedProfiles: boolean
  loading: boolean
  stringifiedEncryptionKey: undefined | StringifiedKeyWithSalt
  enteredWrongPassword: boolean
}

export interface RootStateWithoutPersistSlice extends StrictOmit<RootState, 'persist'> {
  persist: null
}

export interface SyncedRootState extends Pick<RootState, 'theme' | 'wallet' | 'network'> {}
