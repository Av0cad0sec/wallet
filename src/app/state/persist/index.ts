import { PayloadAction } from '@reduxjs/toolkit'
import { createSlice } from 'utils/@reduxjs/toolkit'
import { PersistState } from './types'
import { runtimeIs } from 'config'

// eslint-disable-next-line @typescript-eslint/no-unused-vars
import { createPersistedRootReducer } from 'store/reducers'
import { RootState } from 'types'

export const STORAGE_FIELD = 'oasis_wallet_persist_v1'

/** Syncing tabs is only needed in web app, not in extension. */
export const needsSyncingTabs = runtimeIs === 'webapp'

// Simulate with `delete window.BroadcastChannel`
export const isSyncingTabsSupported = typeof BroadcastChannel === 'function'

export function getInitialState(): PersistState {
  return {
    hasPersistedProfiles: !!localStorage.getItem(STORAGE_FIELD),
    // Disable persistence if tabs would override each other.
    isPersistenceUnsupported: needsSyncingTabs && !isSyncingTabsSupported,
    loading: false,
    stringifiedEncryptionKey: undefined,
    enteredWrongPassword: false,
  }
}

const persistSlice = createSlice({
  name: 'persist',
  initialState: getInitialState(),
  reducers: {
    setWrongPassword(state) {
      state.enteredWrongPassword = true
      state.loading = false
    },
    setUnlockedRootState(state, action: PayloadAction<{ rootState: RootState }>) {
      /**
       * Handled in {@link createPersistedRootReducer}.
       * Sets `state.loading = false` and `state.stringifiedEncryptionKey`.
       */
      return
    },

    // Handled in saga
    // ---------------
    /**
     * Set a password to encrypt state with and begin storing it to localStorage.
     */
    setPasswordAsync(state, action: PayloadAction<{ password: string }>) {
      state.loading = true
    },
    /**
     * Load encrypted state from localStorage and decrypt, or set {@link PersistState.enteredWrongPassword}.
     */
    unlockAsync(state, action: PayloadAction<{ password: string }>) {
      state.loading = true
    },
    /**
     * Remove everything from state, especially {@link PersistState.stringifiedEncryptionKey}.
     * UnlockGate should stop any components relying on missing state.
     */
    lockAsync(state) {
      state.loading = true
    },
    /**
     * Remove encrypted state from localStorage and reload.
     */
    eraseAsync(state) {
      state.loading = true
    },
  },
})

export const persistActions = persistSlice.actions

export default persistSlice.reducer
