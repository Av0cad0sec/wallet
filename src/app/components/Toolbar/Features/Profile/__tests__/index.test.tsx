import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { MemoryRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import { configureAppStore } from 'store/configureStore'
import { ThemeProvider } from 'styles/theme/ThemeProvider'
import { Wallet } from 'app/state/wallet/types'
import { PersistState } from 'app/state/persist/types'
import { Profile } from '../'

const closeHandler = jest.fn()
const mockNavigate = jest.fn()
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockNavigate,
}))
// needed to test Trans component
jest.unmock('react-i18next')

const renderComponent = (store: any) =>
  render(
    <Provider store={store}>
      <MemoryRouter>
        <ThemeProvider>
          <Profile closeHandler={closeHandler} />
        </ThemeProvider>
      </MemoryRouter>
    </Provider>,
  )

describe('<Profile  />', () => {
  let store: ReturnType<typeof configureAppStore>

  const unlockedProfile = {
    wallet: {
      selectedWallet: 'dummy',
      wallets: {
        dummy: {
          address: 'dummy',
        } as Wallet,
      },
    },
    persist: {
      hasPersistedProfiles: true,
      stringifiedEncryptionKey: 'unlockedProfile',
    } as PersistState,
  }

  beforeEach(() => {
    store = configureAppStore(unlockedProfile)
  })

  it('should render unavailable state', () => {
    const { container } = renderComponent(store)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render unavailable state', async () => {
    renderComponent(configureAppStore({}))
    expect(screen.getByText(/You can setup your profile while/)).toBeInTheDocument()
    expect(screen.queryByText('Update Profile')).not.toBeInTheDocument()
    await userEvent.click(screen.getByRole('button', { name: 'opening a wallet' }))
    expect(mockNavigate).toHaveBeenCalledWith('/open-wallet')
    expect(closeHandler).toHaveBeenCalled()
  })
})
