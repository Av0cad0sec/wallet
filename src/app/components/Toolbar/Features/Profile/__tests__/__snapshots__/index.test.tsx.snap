// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile  /> should render unavailable state 1`] = `
.c10 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 24px;
  height: 24px;
  fill: #666666;
  stroke: #666666;
}

.c10 g {
  fill: inherit;
  stroke: inherit;
}

.c10 *:not([stroke])[fill='none'] {
  stroke-width: 0;
}

.c10 *[stroke*='#'],.c10 *[STROKE*='#'] {
  stroke: inherit;
  fill: none;
}

.c10 *[fill-rule],
.c10 *[FILL-RULE],
.c10 *[fill*='#'],.c10 *[FILL*='#'] {
  fill: inherit;
  stroke: none;
}

.c0 {
  font-family: Rubik,sans-serif;
  font-size: 18px;
  line-height: 24px;
  background-color: #ffffff;
  color: #16213E;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-width: 384px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-bottom: 12px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  border-bottom: solid 1px rgba(0,0,0,0.33);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-top: 24px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  margin-top: 24px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-top: 48px;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c19 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c15 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  height: 12px;
}

.c20 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  width: 12px;
}

.c14 {
  font-size: 18px;
  line-height: 24px;
}

.c9 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  outline: none;
  border: none;
  padding: 0;
  text-align: inherit;
  line-height: 0;
  padding: 12px;
}

.c9:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #00A9FF;
}

.c9:focus > circle,
.c9:focus > ellipse,
.c9:focus > line,
.c9:focus > path,
.c9:focus > polygon,
.c9:focus > polyline,
.c9:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #00A9FF;
}

.c9:focus::-moz-focus-inner {
  border: 0;
}

.c9:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c9:focus:not(:focus-visible) > circle,.c9:focus:not(:focus-visible) > ellipse,
.c9:focus:not(:focus-visible) > line,.c9:focus:not(:focus-visible) > path,
.c9:focus:not(:focus-visible) > polygon,.c9:focus:not(:focus-visible) > polyline,
.c9:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c9:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

.c12 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 2px solid #0500e2;
  border-radius: 4px;
  color: #16213E;
  padding: 4px 22px;
  font-size: 18px;
  line-height: 24px;
  background-color: #0500e2;
  color: #F8F8F8;
  border-radius: 4px;
  -webkit-transition-property: color,background-color,border-color,box-shadow;
  transition-property: color,background-color,border-color,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
  font-weight: bold;
}

.c12:hover {
  box-shadow: 0px 0px 0px 2px #0500e2;
}

.c12:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #00A9FF;
}

.c12:focus > circle,
.c12:focus > ellipse,
.c12:focus > line,
.c12:focus > path,
.c12:focus > polygon,
.c12:focus > polyline,
.c12:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #00A9FF;
}

.c12:focus::-moz-focus-inner {
  border: 0;
}

.c12:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c12:focus:not(:focus-visible) > circle,.c12:focus:not(:focus-visible) > ellipse,
.c12:focus:not(:focus-visible) > line,.c12:focus:not(:focus-visible) > path,
.c12:focus:not(:focus-visible) > polygon,.c12:focus:not(:focus-visible) > polyline,
.c12:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c12:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

.c16 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 2px solid #d24c00;
  border-radius: 4px;
  color: #16213E;
  padding: 4px 22px;
  font-size: 18px;
  line-height: 24px;
  background-color: #d24c00;
  color: #F8F8F8;
  border-radius: 4px;
  -webkit-transition-property: color,background-color,border-color,box-shadow;
  transition-property: color,background-color,border-color,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
  font-weight: bold;
}

.c16:hover {
  box-shadow: 0px 0px 0px 2px #d24c00;
}

.c16:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #00A9FF;
}

.c16:focus > circle,
.c16:focus > ellipse,
.c16:focus > line,
.c16:focus > path,
.c16:focus > polygon,
.c16:focus > polyline,
.c16:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #00A9FF;
}

.c16:focus::-moz-focus-inner {
  border: 0;
}

.c16:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c16:focus:not(:focus-visible) > circle,.c16:focus:not(:focus-visible) > ellipse,
.c16:focus:not(:focus-visible) > line,.c16:focus:not(:focus-visible) > path,
.c16:focus:not(:focus-visible) > polygon,.c16:focus:not(:focus-visible) > polyline,
.c16:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c16:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

.c18 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 2px solid #0500e2;
  border-radius: 4px;
  color: #16213E;
  padding: 4px 22px;
  font-size: 18px;
  line-height: 24px;
  -webkit-transition-property: color,background-color,border-color,box-shadow;
  transition-property: color,background-color,border-color,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
  font-weight: bold;
}

.c18:hover {
  box-shadow: 0px 0px 0px 2px #0500e2;
}

.c18:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #00A9FF;
}

.c18:focus > circle,
.c18:focus > ellipse,
.c18:focus > line,
.c18:focus > path,
.c18:focus > polygon,
.c18:focus > polyline,
.c18:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #00A9FF;
}

.c18:focus::-moz-focus-inner {
  border: 0;
}

.c18:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c18:focus:not(:focus-visible) > circle,.c18:focus:not(:focus-visible) > ellipse,
.c18:focus:not(:focus-visible) > line,.c18:focus:not(:focus-visible) > path,
.c18:focus:not(:focus-visible) > polygon,.c18:focus:not(:focus-visible) > polyline,
.c18:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c18:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

.c2 {
  font-size: 18px;
  line-height: 24px;
  max-width: 432px;
}

.c8 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  width: 100%;
  padding: 11px;
  font-weight: 600;
  margin: 0;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
  outline: none;
  border: none;
}

.c8:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #00A9FF;
}

.c8:focus > circle,
.c8:focus > ellipse,
.c8:focus > line,
.c8:focus > path,
.c8:focus > polygon,
.c8:focus > polyline,
.c8:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #00A9FF;
}

.c8:focus::-moz-focus-inner {
  border: 0;
}

.c8::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c8::-moz-placeholder {
  color: #AAAAAA;
}

.c8:-ms-input-placeholder {
  color: #AAAAAA;
}

.c8::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c8::-moz-focus-inner {
  border: none;
  outline: none;
}

.c8:-moz-placeholder,
.c8::-moz-placeholder {
  opacity: 1;
}

.c7 {
  position: relative;
  width: 100%;
}

@media only screen and (max-width:768px) {
  .c4 {
    margin-bottom: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c5 {
    border-bottom: solid 1px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c11 {
    margin-top: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c13 {
    margin-top: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c17 {
    margin-top: 24px;
  }
}

@media only screen and (max-width:768px) {
  .c15 {
    height: 6px;
  }
}

<div
  class="c0"
  style="min-height: 100dvh;"
>
  <div
    class="c1"
  >
    <form
      autocapitalize="off"
      autocomplete="off"
      autocorrect="off"
      spellcheck="false"
    >
      <p
        class="c2"
      >
        <label
          for="password1"
        >
          Set a new password
        </label>
      </p>
      <div
        class="c3"
      >
        <div
          class="c4 "
        >
          <div
            class="c5 "
          >
            <div
              class="c6"
            >
              <div
                class="c7"
              >
                <input
                  autocapitalize="off"
                  autocomplete="off"
                  autocorrect="off"
                  class="c8"
                  id="currentPassword"
                  name="currentPassword"
                  placeholder="Current password"
                  required=""
                  spellcheck="false"
                  type="password"
                  value=""
                />
              </div>
              <button
                aria-label="Show password"
                class="c9"
                type="button"
              >
                <svg
                  aria-label="Hide"
                  class="c10"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17c-2.727 0-6-2.778-6-5s3.273-5 6-5 6 2.778 6 5-3.273 5-6 5zm-1-5a1 1 0 1 0 2 0 1 1 0 0 0-2 0zm9-7L4 19"
                    fill="none"
                    stroke="#000"
                    stroke-width="2"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="c3"
      >
        <div
          class="c4 "
        >
          <div
            class="c5 "
          >
            <div
              class="c6"
            >
              <div
                class="c7"
              >
                <input
                  autocapitalize="off"
                  autocomplete="off"
                  autocorrect="off"
                  class="c8"
                  id="password1"
                  name="password1"
                  placeholder="New password"
                  required=""
                  spellcheck="false"
                  type="password"
                  value=""
                />
              </div>
              <button
                aria-label="Show password"
                class="c9"
                type="button"
              >
                <svg
                  aria-label="Hide"
                  class="c10"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17c-2.727 0-6-2.778-6-5s3.273-5 6-5 6 2.778 6 5-3.273 5-6 5zm-1-5a1 1 0 1 0 2 0 1 1 0 0 0-2 0zm9-7L4 19"
                    fill="none"
                    stroke="#000"
                    stroke-width="2"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="c3"
      >
        <div
          class="c4 "
        >
          <div
            class="c5 "
          >
            <div
              class="c6"
            >
              <div
                class="c7"
              >
                <input
                  autocapitalize="off"
                  autocomplete="off"
                  autocorrect="off"
                  class="c8"
                  id="password2"
                  name="password2"
                  placeholder="Re-enter new password"
                  spellcheck="false"
                  type="password"
                  value=""
                />
              </div>
              <button
                aria-label="Show password"
                class="c9"
                type="button"
              >
                <svg
                  aria-label="Hide"
                  class="c10"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17c-2.727 0-6-2.778-6-5s3.273-5 6-5 6 2.778 6 5-3.273 5-6 5zm-1-5a1 1 0 1 0 2 0 1 1 0 0 0-2 0zm9-7L4 19"
                    fill="none"
                    stroke="#000"
                    stroke-width="2"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="c11"
      >
        <button
          class="c12"
          type="submit"
        >
          Update password
        </button>
      </div>
    </form>
    <div
      class="c13"
    >
      <span
        class="c14"
      >
        Deletion
      </span>
      <div
        class="c15"
      />
      <button
        class="c16"
        type="button"
      >
        Delete profile
      </button>
    </div>
  </div>
  <div
    class="c17"
  >
    <button
      class="c18"
      data-testid="profile-modal-lock-wallet"
      type="button"
    >
      <div
        class="c19"
      >
        <svg
          aria-label="Lock"
          class="c10"
          viewBox="0 0 24 24"
        >
          <path
            d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2m-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2m3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1z"
            fill="#333"
          />
        </svg>
        <div
          class="c20"
        />
        Lock profile
      </div>
    </button>
  </div>
</div>
`;
